p2p/adapters/inproc.go

```diff
@@ -243,13 +256,13 @@ func (self *SimNode) RunProtocol(node *SimNode, rw, rrw p2p.MsgReadWriter, peer
 	log.Trace(fmt.Sprintf("protocol starting on peer %v (connection with %v)", self.Id, id))
 	p := p2p.NewPeer(id.NodeID, id.Label(), []p2p.Cap{})
 	go func() {
-		self.network.DidConnect(self.Id, id)
+		self.peerFeed.Send(&p2p.PeerEvent{Type: p2p.PeerEventTypeAdd, Peer: id.NodeID})
 		err := protocol.Run(p, rw)
 		<-peer.Readyc
 		self.RemovePeer(node.Node())
 		peer.Errc <- err
 		log.Trace(fmt.Sprintf("protocol quit on peer %v (connection with %v broken: %v)", self.Id, id, err))
-		self.network.DidDisconnect(self.Id, id)
+		self.peerFeed.Send(&p2p.PeerEvent{Type: p2p.PeerEventTypeDrop, Peer: id.NodeID})
```

Comment:
surely, we should report which protocol disconnected. I know in p2p, protocol drop implies peer disconnect, yet
it seems clearer since there can be many protocols.